src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -11,14 +11,30 @@
 import java.nio.file.Paths;
 import java.util.zip.Deflater;
 
+/**
+ * Main class, containing the interface for PngEncoder.
+ * PngEncoder is a really fast encoder for PNG images in Java.
+ */
 public class PngEncoder {
+    /**
+     * Compression level 9 is the default.
+     * It produces images with a size comparable to ImageIO.
+     */
     public static int DEFAULT_COMPRESSION_LEVEL = Deflater.BEST_COMPRESSION;
 
     private final BufferedImage bufferedImage;
     private final int compressionLevel;
     private final boolean multiThreadedCompressionEnabled;
     private final PngEncoderSrgbRenderingIntent srgbRenderingIntent;
 
+    /**
+     * Constructs a PngEncoder based on the provided parameters.
+     *
+     * @param bufferedImage input image
+     * @param compressionLevel compression level to be applied (must be between -1 and 9 inclusive)
+     * @param multiThreadedCompressionEnabled when {@code true}, multithreaded compression will be used
+     * @param srgbRenderingIntent todo
```

Comment:
As @johantiden mentioned, it might be unnecessary to document the private constructor.

But if we were to keep this comment it might be a good example of where to follow [the official Javadoc guide's recommendation](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#@param) to use spaces to line up the parameter descriptions:
```
/**
 * Constructs a PngEncoder based on the provided parameters.
 *
 * @param bufferedImage                    input image
 * @param compressionLevel                 compression level to be applied (must be between -1 and 9 inclusive)
 * @param multiThreadedCompressionEnabled  when {@code true}, multithreaded compression will be used
 * @param srgbRenderingIntent              todo
 */
```

Reply:
Thanks for the feedback. That's easier to read.